### PR TITLE
Update service client E2E tests

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: e2e/test/iothub/service/ServiceClientE2ETests.cs [hunk 1]
```diff
@@ -34,29 +35,29 @@ public async Task Message_NoTimeoutPassed()
 
         private async Task FastTimeout()
         {
-            TimeSpan? timeout = TimeSpan.FromTicks(10).Negate();
-            await TestTimeout(timeout).ConfigureAwait(false);
+            TimeSpan timeout = TimeSpan.FromTicks(10).Negate();
+            using var cts = new CancellationTokenSource(timeout);
```
Comment (timtay-microsoft) [inline e2e/test/iothub/service/ServiceClientE2ETests.cs:39]: ```new CancellationToken(true);``` is all you need if you want a pre-cancelled cancellation token

File: e2e/test/iothub/service/ServiceClientE2ETests.cs [hunk 2]
```diff
@@ -34,29 +35,29 @@ public async Task Message_NoTimeoutPassed()
...
+        private async Task TestTimeout(CancellationToken token = default)
         {
             using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, DevicePrefix).ConfigureAwait(false);
             using var sender = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
 
             var sw = new Stopwatch();
             sw.Start();
 
-            Logger.Trace($"Testing ServiceClient SendAsync() timeout in ticks={timeout?.Ticks}");
+            Logger.Trace($"Testing ServiceClient SendAsync() cancellation.");
```
Comment (timtay-microsoft) [inline e2e/test/iothub/service/ServiceClientE2ETests.cs:56]: David has [expressed some interest](https://github.com/Azure/azure-iot-sdk-csharp/pull/2614#discussion_r953150296) in removing these tests

**Review by timtay-microsoft — APPROVED** [review]

File: e2e/test/iothub/service/ServiceClientE2ETests.cs (hunk 2 above)
Comment (abhipsaMisra) [reply e2e/test/iothub/service/ServiceClientE2ETests.cs:56]: There's some value in ensuring that the cancellation token is propagated to the transport layer, but we could probably do that through unit tests as well.

Comment (drwill-ms) [reply e2e/test/iothub/service/ServiceClientE2ETests.cs:56]: Absolutely, should be a unit test to confirm a pre-canceled token aborts the operational call before it is made.

**Pull request merged**